budgetportal/tests/test_provincial_infrastructure_projects.py

```diff
@@ -168,3 +174,141 @@ def test_other_parties(self):
+        self.province = u"Eastern Cape"
+        self.status = u"Construction"
+        self.source = u"Community Library Service Grant"
+        self.department = u"Department 1"
+        self.name = u"Project 1"
+        self.municipality = u"Local 2"
+        self.contractor = u"Contractor 3"
+
+        self.provinces = [
+            "Eastern Cape",
+            "Free State",
+            "Gauteng",
+            "KwaZulu-Natal",
+            "Limpopo",
+            "Mpumalanga",
+            "North West",
+            "Northern Cape",
+            "Western Cape",
+        ]
+        self.statuses = ["Design", "Tender", "Feasibility", "Construction"]
+        self.sources = [
+            "Equitable Share",
+            "Education Infrastructure Grant",
+            "Community Library Service Grant",
+        ]
+        for i in range(30):
+            ProvInfraProject.objects.create(
+                financial_year=self.fin_year,
+                IRM_project_id=i,
+                name="Project {}".format(i),
+                department="Department {}".format(i),
+                local_municipality="Local {}".format(i),
+                district_municipality="District {}".format(i),
+                province=random.choice(self.provinces),
+                status=random.choice(self.statuses),
+                primary_funding_source=random.choice(self.sources),
+                main_contractor="Contractor {}".format(i),
+                principle_agent="Principle Agent {}".format(i),
+                program_implementing_agent="Program Agent {}".format(i),
+                other_parties="Service Provider: XXXX{}".format(i),
+            )
+
+    def test_projects_per_page(self):
+        response = self.client.get(self.url, format="json")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        number_of_results = len(response.data["results"])
+        self.assertLessEqual(number_of_results, 20)
+
+    def test_filter_by_department(self):
+        project = ProvInfraProject.objects.get(department=self.department)
+
+        data = {"department": self.department}
+        response = self.client.get(self.url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        number_of_projects = len(response.data["results"])
+        response_data = response.data["results"][0]
+        self.assertEqual(number_of_projects, 1)
+        self.assertEqual(response_data["name"], project.name)
+        self.assertEqual(response_data["province"], project.province)
+        self.assertEqual(
+            response_data["local_municipality"], project.local_municipality
+        )
+        self.assertEqual(
+            response_data["district_municipality"], project.district_municipality
+        )
+
+    def test_filter_by_province(self):
+        projects = ProvInfraProject.objects.filter(province=self.province)
+
+        data = {"province": self.province}
```

Comment:
again, just write "Eastern Cape" here. or maybe put it in a variable inside the test method because you're using it twice. But let the literal be visible.

Please do that for all of these search values.

Reply:
variable defined.

https://github.com/vulekamali/datamanager/pull/190/commits/ee17bc29b03185e791fa6fa2001619b1076a4683